### PR TITLE
Adjust Button spacing and use ButtonLink in menu

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
- Use ButtonNextLink in NavigationTrigger
- Update text alignment in Button to be more visually centered according to new UI kit

Screenshot 2024-07-15 at 11 28 25

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
* **#4485** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4485?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gustaveen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/ui/src/components/Button/Button.css.ts [hunk 1]
```diff
@@ -230,6 +230,12 @@ export const childrenWrapper = style({
   gap: tokens.space.xs,
 })
 
+export const textWrapper = style({
```
Comment (alebedev) [inline packages/ui/src/components/Button/Button.css.ts:233]: I just needed this in PriceCalculator. Will try to find more universal solution

**Review by alebedev — APPROVED** [review]
Nice!

File: packages/ui/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -71,7 +78,7 @@ export const Button: PolymorphicComponent = forwardRef(function Button<
       {...props}
     >
       <span className={childrenWrapper} style={{ opacity: loading ? 0 : 1 }}>
```
[the remark this answers is not in the document]
Comment (gustaveen) [reply packages/ui/src/components/Button/Button.tsx:80]: For sure, let's fix that 🙌

**Comment by gustaveen** [comment]
### Merge activity

* **Jul 16, 4:10 AM EDT**: @gustaveen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4485).

**Pull request merged**